x-pack/docs/en/security/authentication/saml-guide.asciidoc

```diff
@@ -857,3 +857,145 @@ A conscious effort has been made to mask this complexity with sane defaults and
+
+On a high level, the custom web application would need to perform the
+following steps in order to authenticate a user with SAML against {es}
+
+. Make an HTTP POST request to `_security/saml/prepare`, authenticating as
+the `saml-service-user` user, using either the name of the SAML realm in the {es} configuration or the value for
+the Assertion Consumer Service URL in the request body.
+See the {ref}/security-api-saml-prepare-authentication.html[SAML Prepare Authentication API] for more details
++
+[source,js]
+--------------------------------------------------
+POST /_security/saml/prepare
+{
+  "realm" : "saml1"
+}
+--------------------------------------------------
+// CONSOLE
+
+. Handle the response from `/_security/saml/prepare`. The response from {es} will contain 3 parameters:
+  `redirect`, `realm' and `id`. The custom web application would need to store the value for `id`
+ in the user's session (client side in a cookie or server side if session information is
+persisted this way) and redirect the user's browser to the URL that will be contained in the
+  `redirect` value. The `id` value should not be disregarded as it is used as a nonce in SAML in
+order to mitigate against replay attacks.
+. Handle a subsequent response from the SAML IdP. After the user is successfully authenticated with the
+Identity Provider they will be redirected back to the Assertion Consumer Service URL. This (`sp.acs`) needs to be
+defined as a URL which the custom we application handles. Upon receiving this HTTP POST request, the
+custom web application will need to parse that and make an HTTP POST request itself to the
+`_security/saml/authenticate` API, again - authenticating as the `saml-service-user` user - passing
+the Base64 encoded SAML Response that was sent as the body of the request as well as the value for `id` that it had saved in the user's session previously.
++
+See {ref}/security-api-saml-authenticate.html[SAML Authenticate API] for more details
++
+[source,js]
+-----------------------------------------------------------------------
+POST /_security/saml/authenticate
+{
+  "content" : "PHNhbWxwOlJlc3BvbnNlIHhtbG5zOnNhbWxwPSJ1cm46b2FzaXM6bmFtZXM6dGM6U0FNTDoyLjA6cHJvdG9jb2wiIHhtbG5zOnNhbWw9InVybjpvYXNpczpuYW1lczp0YzpTQU1MOjIuMD.....",
+  "ids" : ["4fee3b046395c4e751011e97f8900b5273d56685"]
+}
+-----------------------------------------------------------------------
+// CONSOLE
+// TEST[skip:handled in IT]
++
+Elasticsearch will validate this and if all is correct will respond with an access token that can be used
+as a `Bearer` token for subsequent requests and a refresh token that can be later used to refresh the given
+access token as described in {ref}/security-api-get-token.html[get token API].
+. The response to calling `/_security/saml/authenticate` will only contain the username of the authenticated
+user. If you need to get the values for the SAML Attributes for that user that were contained in the SAML
```

Comment:
It is unclear to me what some of the "that" occurrences in this sentence refer to. Maybe:

If you need to get the values for the SAML attributes that were returned in the SAML response for that user, you can call the authenticate API. Use the access token....